community/kernel/src/main/java/org/neo4j/kernel/api/AccessLevel.java

```diff
@@ -52,8 +55,17 @@ public boolean allowsSchemaWrites()
         {
             return true;
         }
+
+        @Override
+        public boolean mayLimitAccessTo( AccessLevel newMode ) { return true; }
     };
 
     public abstract boolean allowsWrites();
     public abstract boolean allowsSchemaWrites();
+    public abstract boolean mayLimitAccessTo( AccessLevel newMode );
+
+    public interface Reset
```

Comment:
Does not sound like a good place for something that will reset level in KernelTransaction, looks strange and suspicious